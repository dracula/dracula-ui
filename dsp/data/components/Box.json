{
  "class": "component",
  "type": "page",
  "id": "Box",
  "name": "Box",
  "last_updated": "2021-03-14T00:08:30.329Z",
  "last_updated_by": "System",
  "description": "Box is the most primitive component of Dracula UI.\nUsing Box allows for consumers of the components library to compose\nmore complex patterns, components, and UIs.\n\nBox includes built-in Color and Spacing properties that make building\ncomplex components convenient and consistent.",
  "related_entity_ids": [],
  "tags": ["component"],
  "snippets": {
    "trigger": "box",
    "languages": {
      "javascriptreact": "<Box color=\"purple\">Dracula</Box>;\n",
      "html": "<div color=\"purple\" class=\"drac-box drac-bg-purple\">Dracula</div>\n"
    }
  },
  "ext_com_draculaui_variations": [
    {
      "title": "Usage",
      "html": "<div color=\"purple\" class=\"drac-box drac-bg-purple\">Dracula</div>\n",
      "react": "<Box color=\"purple\">Dracula</Box>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/Box.png",
      "svgData": "./dsp/assets/svgs/Box.svg"
    },
    {
      "title": "Colors",
      "html": "<div color=\"pinkPurple\" class=\"drac-box drac-bg-pink-purple\">Pink Purple</div>\n",
      "react": "<Box color=\"pinkPurple\">Pink Purple</Box>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxColors.png",
      "svgData": "./dsp/assets/svgs/BoxColors.svg"
    },
    {
      "title": "Rounded",
      "html": "<div style=\"display: flex; flex-direction: row\">\n  <div\n    color=\"purple\"\n    rounded=\"none\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-none\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"sm\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-sm\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"base\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"lg\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-lg\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"2xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-2xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"3xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-3xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"full\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-full\"\n  ></div>\n</div>\n",
      "react": "<div style={{ display: \"flex\", flexDirection: \"row\" }}>\n  <Box\n    color=\"purple\"\n    rounded=\"none\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"sm\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"base\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"lg\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"2xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"3xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"full\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxRounded.png",
      "svgData": "./dsp/assets/svgs/BoxRounded.svg"
    },
    {
      "title": "Generic Elements",
      "html": "<div>\n  <button as=\"button\" color=\"pinkPurple\" class=\"drac-box drac-bg-pink-purple\">\n    Button</button\n  ><a as=\"a\" color=\"green\" href=\"#\" class=\"drac-box drac-bg-green\">Link</a>\n</div>\n",
      "react": "<div>\n  <Box as=\"button\" color=\"pinkPurple\">\n    Button\n  </Box>\n  <Box as=\"a\" color=\"green\" href=\"#\">\n    Link\n  </Box>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxGeneric Elements.png",
      "svgData": "./dsp/assets/svgs/BoxGeneric Elements.svg"
    }
  ],
  "ext_com_draculaui_props": {
    "color": {
      "defaultValue": null,
      "description": "The background color.",
      "name": "color",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": {
        "name": "\"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
      }
    },
    "rounded": {
      "defaultValue": null,
      "description": "The border radius.",
      "name": "rounded",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"base\" | \"lg\" | \"xl\" | \"2xl\" | \"3xl\" | \"full\" | undefined"
      }
    },
    "spacing": {
      "defaultValue": null,
      "description": "Dracula UI standard spacing properties.",
      "name": "spacing",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "SpacingPropType | undefined" }
    },
    "margin": {
      "defaultValue": null,
      "description": "Dracula UI standard margin properties.",
      "name": "margin",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "MarginPropType | undefined" }
    },
    "as": {
      "defaultValue": null,
      "description": "The HTML element to be used",
      "name": "as",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "keyof HTMLElementTagNameMap | undefined" }
    }
  },
  "ext_com_draculaui_docgen": {
    "tags": {},
    "description": "Box is the most primitive component of Dracula UI.\nUsing Box allows for consumers of the components library to compose\nmore complex patterns, components, and UIs.\n\nBox includes built-in Color and Spacing properties that make building\ncomplex components convenient and consistent.",
    "displayName": "Box",
    "methods": [],
    "props": {
      "color": {
        "defaultValue": null,
        "description": "The background color.",
        "name": "color",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": {
          "name": "\"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
        }
      },
      "rounded": {
        "defaultValue": null,
        "description": "The border radius.",
        "name": "rounded",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"base\" | \"lg\" | \"xl\" | \"2xl\" | \"3xl\" | \"full\" | undefined"
        }
      },
      "spacing": {
        "defaultValue": null,
        "description": "Dracula UI standard spacing properties.",
        "name": "spacing",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "SpacingPropType | undefined" }
      },
      "margin": {
        "defaultValue": null,
        "description": "Dracula UI standard margin properties.",
        "name": "margin",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "MarginPropType | undefined" }
      },
      "as": {
        "defaultValue": null,
        "description": "The HTML element to be used",
        "name": "as",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "keyof HTMLElementTagNameMap | undefined" }
      }
    }
  }
}
