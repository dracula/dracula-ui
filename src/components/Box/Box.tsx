import cx from 'classnames/dedupe'
import React, { AllHTMLAttributes } from 'react'
import { borderColors, colors, glowColors } from '../../base/colors'
import {
  PaddingMixin,
  paddingMixin,
  marginMixin,
  MarginMixin
} from '../../base/spacing'

type Element = HTMLElementTagNameMap

export const roundedBorders = {
  none: 'drac-rounded-none',
  sm: 'drac-rounded-sm',
  base: 'drac-rounded',
  lg: 'drac-rounded-lg',
  xl: 'drac-rounded-xl',
  '2xl': 'drac-rounded-2xl',
  '3xl': 'drac-rounded-3xl',
  full: 'drac-rounded-full'
}

/**
 * Box Props
 */
export type BoxProps<K extends keyof Element = 'div'> = {
  /** The background color. */
  color?: keyof typeof colors

  /** The glow color. */
  glowColor?: keyof typeof glowColors

  /** The border color. */
  borderColor?: keyof typeof borderColors

  /** The border radius. */
  rounded?: keyof typeof roundedBorders

  /** The HTML element to be used */
  as?: K
} & AllHTMLAttributes<K> &
  PaddingMixin &
  MarginMixin

/**
 * Box is the most primitive component of Dracula UI.
 * Using Box allows for consumers of the components library to compose
 * more complex patterns, components, and UIs.
 *
 * Box includes built-in Color and Spacing properties that make building
 * complex components convenient and consistent.
 */
export function Box<T extends keyof Element>(props: BoxProps<T>) {
  const finalProps = {
    ...props,
    className: cx(
      `drac-box`,
      props.className,
      props.color && colors[props.color],
      props.glowColor && glowColors[props.glowColor],
      props.borderColor && borderColors[props.borderColor],
      props.rounded && roundedBorders[props.rounded],
      ...paddingMixin(props),
      ...marginMixin(props)
    )
  }

  const as = finalProps.as ?? 'div'
  return React.createElement(as, finalProps, props.children)
}

Box.displayName = 'Box'
