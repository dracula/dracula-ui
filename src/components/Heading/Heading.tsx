import cx from 'classnames/dedupe'
import mapValues from 'lodash/mapValues'
import React, { HTMLAttributes } from 'react'
import { colors } from '../../base/colors'
import {
  cleanProps,
  marginMixin,
  MarginMixin,
  paddingMixin,
  PaddingMixin
} from '../../base/spacing'

export const headingSizes = {
  '2xl': 'drac-heading-2xl',
  xl: 'drac-heading-xl',
  lg: 'drac-heading-lg',
  md: 'drac-heading',
  sm: 'drac-heading-sm',
  xs: 'drac-heading-xs'
}

export const headingColors = mapValues(colors, (className) => {
  return className.replace('-bg-', '-text-')
})

/** Heading Props */
export interface HeadingProps
  extends HTMLAttributes<HTMLHeadingElement>,
    MarginMixin,
    PaddingMixin {
  /**
   * The size and type of Heading to be used.
   * Denotes hierarchy.
   */
  size?: keyof typeof headingSizes

  /**
   * The Dracula UI color to be applied to the Heading.
   */
  color?: keyof typeof headingColors

  /** The HTML element to be used */
  as?: keyof HTMLElementTagNameMap
}

/**
 * Heading is used to display headlines and other forms of hierarchical Text.
 *
 * Headings are similar to the base Text component, but restricted to certain
 * sizes and font weight.
 */
export const Heading: React.FC<HeadingProps> = (props: HeadingProps) => {
  const { size, color, as, ...htmlProps } = props

  const finalProps = {
    ...htmlProps,
    className: cx(
      `drac-heading`,
      props.className,
      headingSizes[size ?? 'xl'],
      headingColors[color ?? 'white'],
      ...paddingMixin(props),
      ...marginMixin(props)
    )
  }

  return React.createElement(as ?? 'h2', cleanProps(finalProps), props.children)
}

Heading.displayName = 'Heading'
