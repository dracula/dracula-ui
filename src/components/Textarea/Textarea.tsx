import cx from 'classnames/dedupe'
import React, { TextareaHTMLAttributes } from 'react'
import { BaseColorMap } from '../../base/colors'
import {
  cleanProps,
  marginMixin,
  MarginMixin,
  paddingMixin,
  PaddingMixin
} from '../../base/spacing'

export const textareaVariants = {
  normal: 'drac-textarea',
  outline: 'drac-textarea-outline'
}

export const textareaSizes = {
  lg: 'drac-textarea-lg',
  md: 'drac-textarea',
  sm: 'drac-textarea-sm'
}

export const textareaBorderSizes = {
  lg: 'drac-textarea-border-lg',
  md: 'drac-textarea-border-md',
  sm: 'drac-textarea-border-sm'
}

export const textareaColors: BaseColorMap & { white: string } = {
  white: 'drac-textarea-white drac-text-white',
  cyan: 'drac-textarea-cyandrac-text-cyan',
  green: 'drac-textarea-green drac-text-green',
  orange: 'drac-textarea-orange drac-text-orange',
  pink: 'drac-textarea-pink drac-text-pink',
  purple: 'drac-textarea-purple drac-text-purple',
  red: 'drac-textarea-red drac-text-red',
  yellow: 'drac-textarea-yellow drac-text-yellow'
}

export interface TextareaProps 
  extends Omit<TextareaHTMLAttributes<HTMLTextAreaElement>, 'size'>, 
  PaddingMixin, 
  MarginMixin {
  
  color?: keyof typeof textareaColors

  size?: keyof typeof textareaSizes | number

  borderSize?: keyof typeof textareaBorderSizes

  variant?: keyof typeof textareaVariants
}

export const Textarea = React.forwardRef<HTMLTextAreaElement, TextareaProps>(
  (props, ref) => {
    const { color, size, borderSize, variant, ...htmlProps} = props

    const finalProps: Record<string, any> = {
      ...htmlProps,
	  className: cx(
	    `drac-textarea`,
	    props.className,
	    variant && textareaVariants[variant],
	    size && typeof size === 'string' && textareaSizes[size],
	    borderSize && textareaBorderSizes[borderSize],
	    color && textareaColors[color],
	    ...paddingMixin(props),
	    ...marginMixin(props)
	  )
    }

    if (size && typeof size === 'number') {
	  finalProps[size] = size
    }

    return <textarea ref={ref} {...cleanProps(finalProps)}></textarea>
  }
)

Textarea.displayName = 'Textarea'
