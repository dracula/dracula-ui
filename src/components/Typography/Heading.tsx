import cx from 'classnames'
import { mapValues } from 'lodash'
import React, { HTMLAttributes } from 'react'
import { colors } from '../../base/colors'
import { spacingClasses, SpacingPropType } from '../../base/spacing'

export const headingSizes = {
  'heading-1': 'drac-heading-1',
  'heading-2': 'drac-heading-2',
  'heading-3': 'drac-heading-3',
  'heading-4': 'drac-heading-4',
  'heading-5': 'drac-heading-5',
  'heading-6': 'drac-heading-6'
}

export const headingColors = mapValues(colors, (className) => {
  return className.replace('-bg-', '-text-')
})

/** Heading Props */
export interface HeadingProps extends HTMLAttributes<HTMLHeadingElement> {
  /**
   * The size and type of Heading to be used.
   * Denotes hierarchy.
   */
  size?: keyof typeof headingSizes

  /**
   * The Dracula UI color to be applied to the Heading.
   */
  color?: keyof typeof headingColors

  /**
   * Controls the spacing between the Heading component and its parent and siblings.
   */
  spacing?: SpacingPropType
}

/**
 * Heading is used to display headlines and other forms of hierarchical Text.
 *
 * Headings are similar to the base Text component, but restricted to certain
 * sizes and font weight.
 */
export const Heading: React.FC<HeadingProps> = (props: HeadingProps) => {
  const tag = {
    'heading-1': 'h1',
    'heading-2': 'h2',
    'heading-3': 'h3',
    'heading-4': 'h4',
    'heading-5': 'h5',
    'heading-6': 'h6'
  }

  const size = tag[props.size ?? 'heading-1']
  const finalProps = {
    ...props,
    className: cx(
      `drac-heading`,
      headingSizes[props.size ?? 'heading-1'],
      headingColors[props.color ?? 'white'],
      ...spacingClasses(props.spacing ?? 'none')
    )
  }

  return React.createElement(size, finalProps, props.children)
}

Heading.displayName = 'Heading'
