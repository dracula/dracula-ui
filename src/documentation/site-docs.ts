import {
  ComponentExample,
  renderSnapshot,
  SnapshotBuilder
} from './render-component'
import React from 'react'
import fs from 'fs-extra'
import { pretty } from './pretty'
import { getDocGen } from '../documentation/docgen/doc-generator'
import { toDSP } from '../documentation/dsp/component-generator'
import puppeteer from 'puppeteer'
import { componentScreenshot } from './screenshots/component-screenshot'
import os from 'os'

interface Documentation {
  basic: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder> | SnapshotBuilder
}

jest.setTimeout(10000)

const RECORDING = process.env.RECORDING

export function docs<T>(
  componentClass: React.FunctionComponent<T> | string,
  documentation: Documentation
) {
  const name =
    typeof componentClass === 'string'
      ? componentClass
      : componentClass.displayName ?? componentClass.name

  const dspPath = `${process.cwd()}/dsp`

  describe(`Site: ${name}`, () => {
    let browser: puppeteer.Browser

    beforeAll(async () => {
      if (!RECORDING) {
        return
      }

      let executablePath = undefined
      if (os.arch() === 'arm64') {
        executablePath = '/Applications/Chromium.app/Contents/MacOS/Chromium'
      }

      browser = await puppeteer.launch({
        headless: true,
        devtools: true,
        executablePath
      })
    })

    let examples: Record<string, ComponentExample> = {}

    afterAll(async () => {
      if (!RECORDING) {
        return
      }

      await browser.close()
      const docGen = getDocGen(name)
      const dsp = toDSP(name, examples, docGen)

      await fs.mkdirp(`${dspPath}/data/components/`).catch(() => {
        // folder already exists
      })

      const path = `${dspPath}/data/components/${name}.json`
      await fs.writeFile(path, pretty(JSON.stringify(dsp), 'json'))
    })

    test('Basic Usage', async () => {
      const variation = documentation.basic()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp,
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()

      if (RECORDING) {
        const [screenshot, svg] = await componentScreenshot(
          browser,
          snapshot,
          null,
          name
        )

        snapshot.screenshot = screenshot
        snapshot.svgData = svg
      }

      examples = {
        ...examples,
        basic: snapshot
      }
    })

    const variations: SnapshotBuilder[] = ([] as SnapshotBuilder[]).concat(
      documentation.variations()
    )

    variations.forEach((variation) => {
      test(variation.title, async () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp,
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()

        if (RECORDING) {
          const [screenshot, svg] = await componentScreenshot(
            browser,
            snapshot,
            variation,
            name
          )

          snapshot.screenshot = screenshot
          snapshot.svgData = svg
        }

        examples = {
          ...examples,
          [variation.title]: snapshot
        }
      })
    })
  })
}
