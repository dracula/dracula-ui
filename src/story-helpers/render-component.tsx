import React from 'react'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { pretty } from './pretty'

Enzyme.configure({ adapter: new Adapter() })

export type ComponentExample = {
  html: string
  docs: string
  title: string
  react?: any
  screenshot?: string
  svgData?: string
}

export type SnapshotBuilder = {
  comp: () => React.ReactElement
  docs: string
  title: string
}

export function buildSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): SnapshotBuilder {
  return {
    title,
    comp,
    docs: docs.trim()
  }
}

export function renderSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): ComponentExample {
  let component = comp()
  const html = mount(component).html()
  let wrapped = false

  // wrap the component in a div when it's not wrapped already
  // we do this so we can get the actual component example to not be shallow rendered.
  // Shallow rendering a parent "div" gives us a nice JSX representation of the example.
  if (component.type !== 'div') {
    component = <div>{component}</div>
    wrapped = true
  }

  let react = shallow(component).debug({ verbose: true })

  if (wrapped) {
    // replace wrapper div ðŸ˜¬
    react = react
      .replace(/^<div>\n/, '')
      .replace(/\n<\/div>$/, '')
      .trim()
  }

  return {
    title,
    html: pretty(html, 'html'),
    react: pretty(react, 'typescript'),
    docs
  }
}
