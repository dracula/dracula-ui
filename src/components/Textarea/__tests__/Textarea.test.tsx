import React from 'react'
import { docs } from '@/documentation/site-docs'
import { snapshot } from '@/documentation/render-component'
import { Box } from '@/components/Box/Box'
import { Textarea } from '@/components/Textarea/Textarea'
import { Heading } from '@/components/Heading/Heading'

docs(Textarea, {
  basic() {
    return snapshot('Usage', Usage, 'Styles or abstrats HTML textarea elements.')
  },
  variations() {
  	return [
  	  snapshot(
  	  	'Colors',
  	  	Colors,
  	  	'Textareas can be customized to use any of the Dracula UI theme colors.'
  	  ),
  	  snapshot(
  	  	'Sizes',
  	  	Sizes,
  	  	'Textareaas can be customized to use several different sizes.'
  	  ),
  	  snapshot(
  	  	'Variants',
  	  	Variants,
  	  	'Use the `outline` variant to represent subtler Textareas'
  	  ),
  	  snapshot(
  	  	'Border Sizes',
  	  	BorderSizes,
  	  	'Border size can be customized to use several different sizes'
  	  )
  	]
  }
})

function Usage() {
	return <Textarea placeholder="Textarea" color="white"></Textarea>
}

function Colors() {
  return (
    <Box>
      <Textarea color="purple" placeholder="purple" m="xs"></Textarea>
      <Textarea color="green" placeholder="green" m="xs"></Textarea>
    </Box>
  )
}

function Sizes() {
  return (
    <Box>
      <Textarea color="white" size="sm" placeholder="small" m="xs"></Textarea>
      <Textarea color="white" size="md" placeholder="medium" m="xs"></Textarea>
      <Textarea color="white" size="lg" placeholder="large" m="xs"></Textarea>
    </Box>
  )
}

function Variants() {
  return (
    <Box>
      <Textarea color="white" variant="normal" placeholder="normal" m="xs"></Textarea>
      <Textarea color="white" variant="outline" placeholder="outline" m="xs"></Textarea>
    </Box>
  )
}

function BorderSizes() {
  return (
    <Box>
      <Box mb="sm">
        <Heading size="sm">sm</Heading>
        <Textarea
          color="green"
          variant="normal"
          borderSize="sm"
          placeholder="small border outline"
          m="xs"
        ></Textarea>
        <Textarea
          color="green"
          variant="outline"
          borderSize="sm"
          placeholder="small border outline"
          m="xs"
        ></Textarea>
      </Box>

      <Box mb="sm">
        <Heading size="sm">md</Heading>

        <Textarea
          color="purple"
          variant="normal"
          borderSize="md"
          placeholder="medium border outline"
          m="xs"
        ></Textarea>
        <Textarea
          color="purple"
          variant="outline"
          borderSize="md"
          placeholder="medium border outline"
          m="xs"
        ></Textarea>
      </Box>

      <Box mb="sm">
        <Heading size="sm">lg</Heading>

        <Textarea
          color="yellow"
          variant="normal"
          borderSize="lg"
          placeholder="large border outline"
          m="xs"
        ></Textarea>
        <Textarea
          color="yellow"
          variant="outline"
          borderSize="lg"
          placeholder="large border outline"
          m="xs"
        ></Textarea>
      </Box>
    </Box>
  )
}
