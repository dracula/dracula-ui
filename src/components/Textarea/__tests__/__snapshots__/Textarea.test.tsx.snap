// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: Textarea Basic Usage 1`] = `
Object {
  "docs": "Styles or abstrats HTML textarea elements.",
  "html": "<textarea
  placeholder=\\"Textarea\\"
  class=\\"drac-textarea drac-textarea-white drac-text-white\\"
></textarea>
",
  "react": "<Textarea placeholder=\\"Textarea\\" color=\\"white\\" />;
",
  "title": "Usage",
}
`;

exports[`Site: Textarea Border Sizes 1`] = `
Object {
  "docs": "Border size can be customized to use several different sizes",
  "html": "<div class=\\"drac-box\\">
  <div class=\\"drac-box drac-mb-sm\\">
    <h2 class=\\"drac-heading drac-heading-sm drac-text-white\\">sm</h2>
    <textarea
      placeholder=\\"small border outline\\"
      class=\\"drac-textarea drac-textarea-border-sm drac-textarea-green drac-text-green drac-m-xs\\"
    ></textarea
    ><textarea
      placeholder=\\"small border outline\\"
      class=\\"drac-textarea drac-textarea-outline drac-textarea-border-sm drac-textarea-green drac-text-green drac-m-xs\\"
    ></textarea>
  </div>
  <div class=\\"drac-box drac-mb-sm\\">
    <h2 class=\\"drac-heading drac-heading-sm drac-text-white\\">md</h2>
    <textarea
      placeholder=\\"medium border outline\\"
      class=\\"drac-textarea drac-textarea-border-md drac-textarea-purple drac-text-purple drac-m-xs\\"
    ></textarea
    ><textarea
      placeholder=\\"medium border outline\\"
      class=\\"drac-textarea drac-textarea-outline drac-textarea-border-md drac-textarea-purple drac-text-purple drac-m-xs\\"
    ></textarea>
  </div>
  <div class=\\"drac-box drac-mb-sm\\">
    <h2 class=\\"drac-heading drac-heading-sm drac-text-white\\">lg</h2>
    <textarea
      placeholder=\\"large border outline\\"
      class=\\"drac-textarea drac-textarea-border-lg drac-textarea-yellow drac-text-yellow drac-m-xs\\"
    ></textarea
    ><textarea
      placeholder=\\"large border outline\\"
      class=\\"drac-textarea drac-textarea-outline drac-textarea-border-lg drac-textarea-yellow drac-text-yellow drac-m-xs\\"
    ></textarea>
  </div>
</div>
",
  "react": "<Box>
  <Box mb=\\"sm\\">
    <Heading size=\\"sm\\">sm</Heading>
    <Textarea
      color=\\"green\\"
      variant=\\"normal\\"
      borderSize=\\"sm\\"
      placeholder=\\"small border outline\\"
      m=\\"xs\\"
    />
    <Textarea
      color=\\"green\\"
      variant=\\"outline\\"
      borderSize=\\"sm\\"
      placeholder=\\"small border outline\\"
      m=\\"xs\\"
    />
  </Box>
  <Box mb=\\"sm\\">
    <Heading size=\\"sm\\">md</Heading>
    <Textarea
      color=\\"purple\\"
      variant=\\"normal\\"
      borderSize=\\"md\\"
      placeholder=\\"medium border outline\\"
      m=\\"xs\\"
    />
    <Textarea
      color=\\"purple\\"
      variant=\\"outline\\"
      borderSize=\\"md\\"
      placeholder=\\"medium border outline\\"
      m=\\"xs\\"
    />
  </Box>
  <Box mb=\\"sm\\">
    <Heading size=\\"sm\\">lg</Heading>
    <Textarea
      color=\\"yellow\\"
      variant=\\"normal\\"
      borderSize=\\"lg\\"
      placeholder=\\"large border outline\\"
      m=\\"xs\\"
    />
    <Textarea
      color=\\"yellow\\"
      variant=\\"outline\\"
      borderSize=\\"lg\\"
      placeholder=\\"large border outline\\"
      m=\\"xs\\"
    />
  </Box>
</Box>;
",
  "title": "Border Sizes",
}
`;

exports[`Site: Textarea Colors 1`] = `
Object {
  "docs": "Textareas can be customized to use any of the Dracula UI theme colors.",
  "html": "<div class=\\"drac-box\\">
  <textarea
    placeholder=\\"purple\\"
    class=\\"drac-textarea drac-textarea-purple drac-text-purple drac-m-xs\\"
  ></textarea
  ><textarea
    placeholder=\\"green\\"
    class=\\"drac-textarea drac-textarea-green drac-text-green drac-m-xs\\"
  ></textarea>
</div>
",
  "react": "<Box>
  <Textarea color=\\"purple\\" placeholder=\\"purple\\" m=\\"xs\\" />
  <Textarea color=\\"green\\" placeholder=\\"green\\" m=\\"xs\\" />
</Box>;
",
  "title": "Colors",
}
`;

exports[`Site: Textarea Sizes 1`] = `
Object {
  "docs": "Textareaas can be customized to use several different sizes.",
  "html": "<div class=\\"drac-box\\">
  <textarea
    placeholder=\\"small\\"
    class=\\"drac-textarea drac-textarea-sm drac-textarea-white drac-text-white drac-m-xs\\"
  ></textarea
  ><textarea
    placeholder=\\"medium\\"
    class=\\"drac-textarea drac-textarea-white drac-text-white drac-m-xs\\"
  ></textarea
  ><textarea
    placeholder=\\"large\\"
    class=\\"drac-textarea drac-textarea-lg drac-textarea-white drac-text-white drac-m-xs\\"
  ></textarea>
</div>
",
  "react": "<Box>
  <Textarea color=\\"white\\" size=\\"sm\\" placeholder=\\"small\\" m=\\"xs\\" />
  <Textarea color=\\"white\\" size=\\"md\\" placeholder=\\"medium\\" m=\\"xs\\" />
  <Textarea color=\\"white\\" size=\\"lg\\" placeholder=\\"large\\" m=\\"xs\\" />
</Box>;
",
  "title": "Sizes",
}
`;

exports[`Site: Textarea Variants 1`] = `
Object {
  "docs": "Use the \`outline\` variant to represent subtler Textareas",
  "html": "<div class=\\"drac-box\\">
  <textarea
    placeholder=\\"normal\\"
    class=\\"drac-textarea drac-textarea-white drac-text-white drac-m-xs\\"
  ></textarea
  ><textarea
    placeholder=\\"outline\\"
    class=\\"drac-textarea drac-textarea-outline drac-textarea-white drac-text-white drac-m-xs\\"
  ></textarea>
</div>
",
  "react": "<Box>
  <Textarea color=\\"white\\" variant=\\"normal\\" placeholder=\\"normal\\" m=\\"xs\\" />
  <Textarea color=\\"white\\" variant=\\"outline\\" placeholder=\\"outline\\" m=\\"xs\\" />
</Box>;
",
  "title": "Variants",
}
`;
